### ISSUE TITLE
Sudoku Solver using CNN

### ISSUE THREAD

**Issue by ashis2004** [abhisheks008/DL-Simplified#856]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : </br>Sudoku Solver using CNN
:red_circle: **Aim** : </br>
Network was able to solve the puzzles with 95% accuracy.
Following is a snippet showing network solving the sudoku.

:red_circle: **Dataset** : (https://www.kaggle.com/bryanpark/sudoku) for this project. </br> 
:red_circle: **Approach** : 
| File      | Description |
| :-----------: | :-----------: |
| **sudoku.ipynb** | notebook for running and testing the project |
| **model.py**   | neural network implementation |
| **report.pdf** | Step by step implementataion and description of project |
**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by ashis2004** [comment]
@abhisheks008 pls assign it me I want to work on it

**Comment by abhisheks008** [comment]
Apart from CNN, what are the other models you can implement for this problem statement? To be a contributor in this repository you need to implement at least 3 models and find out the best fitted model by comparing the implemented models based on their accuracy scores.

**Comment by abhisheks008** [comment]
> Variational Autoencoder (VAE): VAE to generate and solve Sudoku puzzles.
> 
> Recurrent Neural Networks (RNN): RNNs or LSTMs for sequence learning.
> 
> Generative Adversarial Networks (GANs): GANs for generating and solving Sudoku puzzles.
> 
> @abhisheks008 pls assign it to me next time I'll keep in mind

Cool @ashis2004 you can go ahead. Follow the project structure and put arrange all the files as per the structure.

**Comment by abhisheks008** [comment]
> > Variational Autoencoder (VAE): VAE to generate and solve Sudoku puzzles.
> > Recurrent Neural Networks (RNN): RNNs or LSTMs for sequence learning.
> > Generative Adversarial Networks (GANs): GANs for generating and solving Sudoku puzzles.
> > @abhisheks008 pls assign it to me next time I'll keep in mind
> 
> Cool @ashis2004 you can go ahead. Follow the project structure and put arrange all the files as per the structure.

Hi @ashis2004 finish the assigned issue first then only you can start working on this issue.

**Comment by Sanjeev-Kumar78** [comment]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title**: Sudoku Solver using CNN </br>
:red_circle: **Aim**: To develop a deep learning model using Convolutional Neural Networks (CNN) to solve Sudoku puzzles automatically.</br>
:red_circle: **Dataset**: [Kaggle](https://www.kaggle.com/datasets/bryanpark/sudoku/code).</br>
:red_circle: **Approach**: Implement and compare at least three models to find the best-fitted algorithm by checking the accuracy scores. Also, perform exploratory data analysis before creating any model.

**Models to be implemented**:
1. **Baseline CNN**: A simple CNN model to set a baseline.
2. **Improved CNN**: A deeper and more complex CNN architecture.
3. **Hybrid Model (CNN + RNN)**: Combines CNN for feature extraction and RNN for sequence prediction.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute to the Project:**
* Create a separate folder named as the Project Title.
* Inside that folder, there will be four main components:
   * **Images** - To store the required images.
   * **Dataset** - To store the dataset or information/source about the dataset.
   * **Model** - To store the machine learning models created using the dataset.
   * **`requirements.txt`** - This file will contain the required packages/libraries to run the project on other machines.
* Inside the **Model** folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note:**
- The issues will be assigned on a first-come, first-served basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include the issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before starting to contribute.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue:**
- Full name: Sanjeev Kumar
- GitHub Profile Link: https://www.github.com/Sanjeev-Kumar78
- Email ID:sanjeev-kumar78@outlook.com
- Participant ID (if applicable):
- Approach for this Project:
- What is your participant role? (Mention the Open Source program)
**GSSOC'24 Contributor role**

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by abhisheks008** [comment]
Hi @Sanjeev-Kumar78 this issue is opened by another contributor, hence it can't be assigned to you.

**Comment by Sanjeev-Kumar78** [comment]
Okay, no issues.
